protobuf/src/main/scala/magnolify/protobuf/ProtobufType.scala

```diff
@@ -133,7 +133,7 @@ object ProtobufField {
       new ProtobufField[T] {
         override type FromT = tc.FromT
         override type ToT = tc.ToT
-        override val hasOptional: Boolean = tc.hasOptional
+        override def hasOptional: Boolean = tc.hasOptional
```

Comment:
In these implementation cases, isn't it preferable to define these fields as `val` (or maybe `lazy val`?)